bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/monitors/CustomOutputStreamMonitor.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright 2023 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.debug.gdbjtag.openocd.dsf.process.monitors;
+
+import java.io.InputStream;
+import java.nio.charset.Charset;
+
+import org.eclipse.debug.internal.core.OutputStreamMonitor;
+
+/**
+ * Monitors the output stream of a system process and notifies listeners of
+ * additions to the stream.
+ * <p>
+ * The output stream monitor reads system out (or err) via and input stream.
+ * @author Ali Azam Rana
+ *
+ */
+@SuppressWarnings("restriction")
+public class CustomOutputStreamMonitor extends OutputStreamMonitor 
```

Comment:
If there are no custom changes in this, could we use the original OutputStreamMonitor itself?

Reply:
The original OutputStreamMonitor had some weird dependencies that were just cluttering up and I was not able to resolve those.